web/src/stream_edit.ts

```diff
@@ -747,6 +766,39 @@ export function initialize(): void {
         $(".dialog_submit_button").attr("data-stream-id", stream_id);
     });
 
+    $("#channels_overlay_container").on("click", ".reactivate", function (this: HTMLElement, e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        function do_unarchive_stream(): void {
+            const stream_id = Number.parseInt(
+                $(".dialog_submit_button").attr("data-stream-id")!,
+                10,
+            );
+            unarchive_stream(stream_id, $(".stream_change_property_info"));
+        }
+
+        const stream_id = get_stream_id(this);
+        const stream = sub_store.get(stream_id);
+
+        const stream_name_with_privacy_symbol_html = render_inline_decorated_channel_name({stream});
+
+        const html_body = render_settings_reactivation_stream_modal();
+
+        confirm_dialog.launch({
+            html_heading: $t_html(
+                {defaultMessage: "Unarchive <z-link></z-link>?"},
+                {"z-link": () => stream_name_with_privacy_symbol_html},
+            ),
+            id: "unarchive-stream-modal",
+            help_link: "/help/archive-a-channel",
```

Comment:
We might want to change this link.

Reply:
I am not sure whether we would have a new page for unarchiving or just keep the section currently at the bottom of https://zulip.com/help/archive-a-channel.

Reply:
So, I think we can just remove the link for now.